cmd/quadlet/main.go

```diff
@@ -0,0 +1,246 @@
+	}
+
+	dirs := make([]string, 0)
+	if user {
+		if configDir, err := os.UserConfigDir(); err == nil {
+			dirs = append(dirs, path.Join(configDir, "containers/systemd"))
+		}
+	} else {
+		dirs = append(dirs, quadlet.UnitDirAdmin)
+		dirs = append(dirs, quadlet.UnitDirDistro)
+	}
+	return dirs
+}
+
+func loadUnitsFromDir(sourcePath string, units map[string]*systemdparser.UnitFile) {
+	files, err := os.ReadDir(sourcePath)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			Log("Can't read \"%s\": %s", sourcePath, err)
+		}
+		return
+	}
+
+	for _, file := range files {
+		name := file.Name()
+		if units[name] == nil &&
+			(strings.HasSuffix(name, ".container") ||
+				strings.HasSuffix(name, ".volume")) {
+			path := path.Join(sourcePath, name)
+
+			Debug("Loading source unit file %s", path)
+
+			if f, err := systemdparser.ParseUnitFile(path); err != nil {
+				Log("Error loading '%s', ignoring: %s", path, err)
+			} else {
+				units[name] = f
+			}
+		}
+	}
+}
+
+func generateServiceFile(outputPath string, service *systemdparser.UnitFile) error {
+	Debug("writing '%s'", service.Path)
+
+	service.PrependComment("",
+		"Automatically generated by quadlet-generator",
+		"")
+
+	f, err := os.Create(service.Path)
+	if err != nil {
+		return err
+	}
+
+	defer f.Close()
+
+	err = service.Write(f)
+	if err != nil {
+		return err
+	}
+
+	f.Sync()
+
+	return nil
+}
+
+func enableServiceFile(outputPath string, service *systemdparser.UnitFile) {
+	symlinks := make([]string, 0)
+
+	aliases := service.LookupAllStrv(quadlet.InstallGroup, "Alias")
+	for _, alias := range aliases {
+		symlinks = append(symlinks, filepath.Clean(alias))
+	}
+
+	wantedBy := service.LookupAllStrv(quadlet.InstallGroup, "WantedBy")
+	for _, wantedByUnit := range wantedBy {
+		// Only allow filenames, not paths
+		if strings.Index(wantedByUnit, "/") == -1 {
+			symlinks = append(symlinks, fmt.Sprintf("%s.wants/%s", wantedByUnit, service.Filename))
+		}
+	}
+
+	requiredBy := service.LookupAllStrv(quadlet.InstallGroup, "RequiredBy")
+	for _, requiredByUnit := range requiredBy {
+		// Only allow filenames, not paths
+		if strings.Index(requiredByUnit, "/") == -1 {
+			symlinks = append(symlinks, fmt.Sprintf("%s.requires/%s", requiredByUnit, service.Filename))
+		}
+	}
+
+	for _, symlinkRel := range symlinks {
+		target, err := filepath.Rel(path.Dir(symlinkRel), service.Filename)
+		if err != nil {
+			Log("Can't create symlink %s: %s", symlinkRel, err)
+			continue
+		}
+		symlinkPath := path.Join(outputPath, symlinkRel)
+
+		symlinkDir := path.Dir(symlinkPath)
+		os.MkdirAll(symlinkDir, os.ModePerm)
+
+		Debug("Creating symlink %s -> %s", symlinkPath, target)
+		_ = os.Remove(symlinkPath) // overwrite existing symlinks
+		err = os.Symlink(target, symlinkPath)
+		if err != nil {
+			Log("Failed creating symlink %s: %s", symlinkPath, err)
+		}
+	}
+}
+
+func main() {
+	prgname := path.Base(os.Args[0])
+	isUser = strings.Index(prgname, "user") != -1
+
+	flag.Parse()
```

Comment:
I suggest moving the var decls, the flag initializing and main() closer.  It's a bit scattered at the moment and wont' fit on one screen.